flink-core/src/main/java/org/apache/flink/core/plugin/DirectoryBasedPluginFinder.java

```diff
@@ -68,39 +63,38 @@ public DirectoryBasedPluginFinder(Path pluginsRootDir) {
 
 	@Override
 	public Collection<PluginDescriptor> findPlugins() throws IOException {
+
+		if (!Files.isDirectory(pluginsRootDir)) {
+			return Collections.emptyList();
```

Comment:
question: Should we throw here?

Reply:
Good question. No strong preference here so we could change it.